test/chainsaw/testmetrics/metrics/chainsaw-test.yaml

```diff
@@ -0,0 +1,79 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  creationTimestamp: null
+  name: metrics
+spec:
+  steps:
+  - name: step-00
+    try:
+    - apply:
+        file: 00-install.yaml
+  - name: step-01
+    try:
+    - apply:
+        file: badmetric1.yaml
+        expect:
+        - check:
+            ($error != null): true
+    - apply:
+        file: badmetric2.yaml
+        expect:
+        - check:
+            ($error != null): true
+    - apply:
+        file: badmetric3.yaml
+        expect:
+        - check:
+            ($error != null): true
+    - apply:
+        file: goodmetric1.yaml
+    - apply:
+        file: goodmetric2.yaml
+    - apply:
+        file: goodmetric3.yaml
+    - apply:
+        file: goodmetric4.yaml
+    - apply:
+        file: goodmetric5.yaml
+    - apply:
+        file: goodmetric6.yaml
+  - name: step-02
+    try:
+    - assert:
+        file: goodmetric1.yaml
+    - assert:
+        file: goodmetric2.yaml
+    - assert:
+        file: goodmetric3.yaml
+    - assert:
+        file: goodmetric4.yaml
+    - assert:
+        file: goodmetric5.yaml
+    - assert:
+        file: goodmetric6.yaml
+    - error:
```

Comment:
out of curiosity is it necesssary to assert errors if we do have a check statement under apply?

Reply:
It means that if the resource exists in the cluster it is considered an error (kind of negative testing, makes sense to me)

Reply:
I see amazing :)